sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -2365,12 +2366,12 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
               ContentLength; // Required for "Append Data" and "Flush Data".  Must be 0 for "Flush
                              // Data".  Must be the length of the request content in bytes for
                              // "Append Data".
-          Azure::Core::Nullable<std::string>
+          Azure::Core::Nullable<Storage::ContentHash>
               TransactionalContentMd5; // Specify the transactional md5 for the body, to be
                                        // validated by the service.
-          Azure::Core::Nullable<std::string>
-              ContentCrc64; // Specify the transactional crc64 for the body, to be validated by the
-                            // service.
+          Azure::Core::Nullable<Storage::ContentHash>
+              TransactionalContentCrc64; // Specify the transactional crc64 for the body, to be
+                                         // validated by the service.
```

Comment:
Why two separate hashes?

Reply:
I don't think we reached a conclusion that the non-user-facing interface has the requirement of unifying content hash for both CRC64 and Md5.

Reply:
I can live with that. But I still think it's far from satisfactory.